frontend/packages/shipwright-plugin/src/api.ts

```diff
@@ -13,6 +14,12 @@ export const startBuild = async (build: Build): Promise<BuildRun> => {
       generateName: `${build.metadata.name}-`,
       labels: {
         [BUILDRUN_TO_BUILD_REFERENCE_LABEL]: build.metadata.name,
+        ...(resource
+          ? {
+              [BUILDRUN_TO_RESOURCE_MAP_LABEL]:
+                resource.metadata?.labels?.[BUILDRUN_TO_RESOURCE_MAP_LABEL],
+            }
+          : {}),
```

Comment:
You want to override the existing `[BUILDRUN_TO_RESOURCE_MAP_LABEL]` ?

Can we go with this version or do you see an issue if the label is set to build.metadata.name if the resource is defined but doesn't have a BUILDRUN_TO_RESOURCE_MAP_LABEL?

```suggestion
        [BUILDRUN_TO_RESOURCE_MAP_LABEL]:
          resource?.metadata?.labels?.[BUILDRUN_TO_RESOURCE_MAP_LABEL] ||
          build.metadata.name),
```

Reply:
No, we are not overriding the label. We are copying it into the new BR if it's present in the Build.
Otherwise, we are not adding that label to the new BR.
I think since we already have the other label `BUILDRUN_TO_BUILD_REFERENCE_LABEL` that is storing the `build.metadata.name`, it's not required to store the same in this label if there is no corresponding label in Build.